### PR TITLE
Fix potential exception with TokenRejectHandler tokenIds

### PR DESCRIPTION
**Pull request by edwin-greene**
**Description**:

- Removes the potential for an invalid TokenReject tokenId to throw an exception in TokenRejectTransactionHandler.
- Fixes RecordItemBuilder tokenReject to now include token transfers.

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/TokenRejectTransactionHandler.java [hunk 1]
```diff
@@ -21,25 +21,30 @@
...
-        super.addCommonEntityIds(transaction, recordItem);
-        var rejections = recordItem.getTransactionBody().getTokenReject().getRejectionsList();
-        for (var rejection : rejections) {
+    protected void doUpdateTransaction(Transaction transaction, RecordItem recordItem) {
+        if (!entityProperties.getPersist().isTokens() || !recordItem.isSuccessful()) {
+            return;
+        }
+
+        var tokenReject = recordItem.getTransactionBody().getTokenReject();
+        tokenReject.getRejectionsList().forEach(rejection -> {
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/TokenRejectTransactionHandler.java:42]: Why do we need this? The token ID is already added to `entity_transaction` in `EntityRecordItemListener.insertFungibleTokenTransfers()` from the transfer list in the record.

File: hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/domain/RecordItemBuilder.java [hunk 2]
```diff
@@ -1194,6 +1213,11 @@ public Builder<T> transactionBodyWrapper(Consumer<TransactionBody.Builder> consu
             return this;
         }
 
+        public Builder<T> tokenTransferList(TokenTransferList.Builder transferList) {
+            transactionRecord.addTokenTransferLists(transferList);
+            return this;
+        }
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/domain/RecordItemBuilder.java:1219]: Remove this and use `Builder.record(r -> r.addTokenTransferLists())`

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/TokenRejectTransactionHandler.java (hunk 1 above)
Comment (edwin-greene) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/TokenRejectTransactionHandler.java]: Thanks I double checked and you are right, it is not needed. I have removed it.

File: hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/domain/RecordItemBuilder.java (hunk 2 above)
Comment (edwin-greene) [reply hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/parser/domain/RecordItemBuilder.java]: Thanks I have updated this

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/TokenRejectTransactionHandler.java [hunk 3]
```diff
@@ -18,30 +18,19 @@
...
 import com.hedera.mirror.importer.domain.EntityIdService;
+import com.hedera.mirror.importer.parser.record.entity.EntityProperties;
 import jakarta.inject.Named;
 import lombok.RequiredArgsConstructor;
 
 @Named
 @RequiredArgsConstructor
 class TokenRejectTransactionHandler extends AbstractTransactionHandler {
 
+    private final EntityProperties entityProperties;
```
Comment (steven-sheehy) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/TokenRejectTransactionHandler.java:31]: Unused

Comment (edwin-greene) [reply hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/TokenRejectTransactionHandler.java]: Got it, removed

**Review by steven-sheehy — APPROVED** [review]

**Review by jascks — APPROVED** [review]
LGTM

**Review by mgoelswirlds — APPROVED** [review]
LGTM

**Review by xin-hedera — APPROVED** [review]
LGTM

**Pull request merged**